x/incentives/keeper/gauge.go

```diff
@@ -208,6 +208,73 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 	return gauge.Id, nil
 }
 
+// TODO: refactor CreateGauge to use this as well
+// type GaugeParams struct {
+// 	IsPerpetual       bool
+// 	DistributeTo      lockuptypes.QueryCondition
+// 	Coins             sdk.Coins
+// 	StartTime         time.Time
+// 	NumEpochsPaidOver uint64
+// }
+
+// TODO: consider doing a general gauge interface?
+// type GroupGauge struct {
+//  Id                uint64
+//  GaugeParams 	  GaugeParams
```

Comment:
Upon second review, not sure if we need GaugeParams for GroupGauge, it can just store an id and internalGaugesId, we might require couple of fields such as coins but not the entire gauge object